tests/unittests/model/test_electrolyzer_n_inputs_yaml.py

```diff
@@ -0,0 +1,504 @@
+# Copyright (c) 2024, RTE (https://www.rte-france.com)
+#
+# See AUTHORS.txt
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# SPDX-License-Identifier: MPL-2.0
+#
+# This file is part of the Antares project.
+
+import math
+import os
+from pathlib import Path
+
+from andromede.libs.standard import DEMAND_MODEL, GENERATOR_MODEL, NODE_BALANCE_MODEL
+from andromede.libs.standard_sc import (
+    CONVERTOR_MODEL,
+    CONVERTOR_RECEIVE_IN,
+    DECOMPOSE_1_FLOW_INTO_2_FLOW,
+    NODE_BALANCE_MODEL_MOD,
+    TWO_INPUTS_CONVERTOR_MODEL,
+)
+from andromede.model.parsing import parse_yaml_library
+from andromede.model.resolve_library import resolve_library
+from andromede.simulation import OutputValues, TimeBlock, build_problem
+from andromede.study import (
+    ConstantData,
+    DataBase,
+    Network,
+    Node,
+    PortRef,
+    create_component,
+)
+
+"""
+for every following test we have two electrical productions with an electrolyzer converting to a gaz flow
+we always have:
+    first electric production:
+        - p_max = 70
+        - cost = 10
+    second electric production:
+        - p_max = 80
+        - cost = 20
+    gaz production:
+        - p_max = 30
+        - cost = 15
+    first conversion rate:
+        - alpha = 0.7
+    second conversion rate:
+        - alpha = 0.5
+    for a gaz demand of 100
+"""
+
+
+def test_electrolyzer_n_inputs_1(data_dir: Path):
+    """
+    Test with an electrolyzer for each input
+
+    ep1 = electric production 1
+    ep2 = electric production 2
+    ez1 = electrolyzer 1
+    ez2 = electrolyzer 2
+    gp = gaz production
+
+    total gaz production = flow_ep1 * alpha_ez1 + flow_ep2 * alpha_ez2 + flow_gp
+
+    """
+    libs_path = Path(__file__).parents[3] / "src/andromede/libs/"
```

Comment:
Lines 70 to 80 are repeated in all yaml tests, you could define `lib` and `lib_sc` in `conftest.py` as pytest fixtures, similarly as what is done for `data_dir`. If you move your tests to `tests/models` you may need to create a `conftest` file within this directory